scripts/Util.py

```diff
@@ -2673,7 +2673,7 @@ def __init__(self, current):
         elif current.config.avd or not current.config.device:
             endpoint = "tcp -h 127.0.0.1 -p 15001"
         RemoteProcessController.__init__(self, current, endpoint)
-        self.device = current.config.device
+        self.device = current.config.device if current.config.device != "" else None
```

Comment:
This is the actual fix. `supportsDiscovery` implementation below checks device is None, but config default is an empty string.